rafthttp/stream.go

```diff
@@ -38,6 +39,16 @@ const (
 	streamBufSize = 4096
 )
 
+var (
+	errUnsupportedStreamType = fmt.Errorf("unsupported stream type")
+
+	streamSupportMap = map[streamType]*semver.Version{
```

Comment:
i prefer to see [app][2.0-nil].

It is inconsistent if we deprecate one stream type in some version, but i think version-index map has the same problem:

in 2.0: {[2.0][app], [2.1][app]}
in 2.1: {[2.0][app], [2.1][app, appv2, message], [2.2][app, appv2, message]}